catwalk/models/metaicl.py

```diff
@@ -0,0 +1,133 @@
+import collections
+from typing import Dict, Any, List, Tuple, Sequence, Iterator, Union, Mapping
+
+import torch
+from transformers import T5ForConditionalGeneration, GPT2LMHeadModel, \
+    GPT2Tokenizer, T5TokenizerFast
+import numpy as np
+import warnings
+
+from catwalk.task import Task, InstanceFormat, RankClassificationInstance
+from catwalk.models.rank_classification import DecoderOnlyRCModel
+
+
+_Model = Union[T5ForConditionalGeneration, GPT2LMHeadModel]
+_Tokenizer = Union[T5TokenizerFast, GPT2Tokenizer]
+
+
+class MetaICLModel(DecoderOnlyRCModel):
+    """A model that reproduces the formatting and truncation of MetaICL."""
+
+    def __init__(self, pretrained_model_name_or_path: str, max_length_per_example : int = 256, continuation_seperator: str = '\n', example_seperatior='\n\n\n'):
```

Comment:
```suggestion
    def __init__(
        self,
        pretrained_model_name_or_path: str,
        *,
        max_length_per_example: int = 256,
        continuation_seperator: str = '\n',
        example_seperatior: str = '\n\n\n'
    ):
```

Line is too long, some of the formatting is inconsistent, and it was missing a type annotation. I also like to err on the side of making arguments keyword-only (with `*`), because it makes the calling code more readable, and gives more options to extend the API later without breaking backwards compatibility.